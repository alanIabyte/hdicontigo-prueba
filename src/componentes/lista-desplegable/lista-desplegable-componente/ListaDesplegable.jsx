import React, { useState } from "react";
import { v4 } from "uuid";
import PropTypes from "prop-types";
import {
  CajaListaDesplegable,
  Contenedor,
  ContenedorListaDesplegable,
  Etiqueta,
  Flecha,
  FlechaListaDesplegable,
  Opcion,
  OpcionListaDesplegable,
  SeleccionListaDesplegable,
} from "./ListaDesplegable.styled";

const ListaDesplegableComponente = (props) => {
  const { alSeleccionar, config, etiqueta } = props;
  const indiceSeleccionado = config.findIndex(
    (elemento) => elemento.seleccionado
  );
  const [opcionSeleccionada, asignarValorOpcionSeleccionada] = useState(
    indiceSeleccionado > 0 ? indiceSeleccionado : 0
  );
  const [despliegaOpciones, asignarValorDespliegaOpciones] = useState(false);

  const elegirOpcion = (opcion, indice) => {
    asignarValorOpcionSeleccionada(indice);
    asignarValorDespliegaOpciones(false);
    alSeleccionar(opcion);
  };

  const cambiarPosicion = () => {
    asignarValorDespliegaOpciones(!despliegaOpciones);
  };

  return (
    <Contenedor>
      <Etiqueta>{etiqueta}</Etiqueta>
      <CajaListaDesplegable onClick={cambiarPosicion}>
        <ContenedorListaDesplegable>
          <SeleccionListaDesplegable>
            {config[opcionSeleccionada].etiqueta}
          </SeleccionListaDesplegable>
          <FlechaListaDesplegable>
            <Flecha despliegaOpciones={despliegaOpciones} />
          </FlechaListaDesplegable>
        </ContenedorListaDesplegable>
      </CajaListaDesplegable>
      <OpcionListaDesplegable despliegaOpciones={despliegaOpciones}>
        {config.map((opcion, indice) => (
          <Opcion
            key={v4()}
            onClick={() => elegirOpcion(opcion, indice)}
            onKeyDown={() => elegirOpcion(opcion, indice)}
            seleccionado={opcionSeleccionada === indice}
            role="button"
            tabIndex={indice}
          >
            {opcion.etiqueta}
          </Opcion>
        ))}
      </OpcionListaDesplegable>
    </Contenedor>
  );
};

ListaDesplegableComponente.propTypes = {
  alSeleccionar: PropTypes.func,
  config: PropTypes.arrayOf(
    PropTypes.shape({
      valor: PropTypes.string,
      etiqueta: PropTypes.string,
      seleccionado: PropTypes.bool,
    })
  ),
  etiqueta: PropTypes.string,
};

ListaDesplegableComponente.defaultProps = {
  alSeleccionar: () => {},
  config: {},
  etiqueta: "",
};

export default ListaDesplegableComponente;
