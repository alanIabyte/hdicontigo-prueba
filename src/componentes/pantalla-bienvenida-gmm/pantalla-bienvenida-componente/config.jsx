const diccionario = {
  mensajeBienvenida: "Te damos la bienvenida",
  mensajeDeErrorDefault: "Ocurrió un error",
  mensajeAlertaContrasenaCambiada:
    "Tu contraseña ha sido actualizada con éxito.",
  mensajeIngreso: "Ingresar",
  terminos: "Términos y condiciones",
  aviso: "Aviso de privacidad",
  etiquetaReportar: "Reportar un siniestro",
  etiquetaIngresar: "Ingresar",
  etiquetaEntrar: "Entrar",
  etiquetaUsuario: "Teléfono",
  etiquetaContrasena: "Contraseña",
  linkOlvideContrasena: "Olvidé mi contraseña",
  errorUsuario:
    "Teléfono debe tener 10 dígitos o formato de correo electrónico erróneo.",
  alertaLogin: {
    titulo: "Ocurrió un problema",
    // eslint-disable-next-line max-len
    cuerpo:
      "Para seguir usando la aplicación debes iniciar sesión. También puedes reportar un siniestro sin iniciar sesión.",
    boton: "Continuar",
  },
  mensajeRegistro: "Regístrate aquí",
  errores: {
    verificarCampos:
      "Favor de verificar tus datos y que estás seleccionando el tipo de póliza correcta.",
    campoRequerido: "Campo requerido para poder continuar.",
    nombreIncompleto: "Por favor introduce tu nombre completo",
    numeroTarjetaIncompleto: "Por favor introduce el número completo",
    formatoIncorrecto: "Por favor revisa que el formato sea correcto.",
    numeroTarjetaNoCoincide: "Los números de tarjeta no coinciden",
    seleccionarAnioActual: "Introduce una tarjeta que no haya expirado",
    CLABEIncompleta: "La cuenta CLABE debe contener 18 dígitos",
    CLABENoCoincide: "Los números de CLABE no coinciden",
  },
};

const configAlertaGeolocation = {
  textoEncabezado: "Geolocalizacion",
  tipoIcono: "ubicacion",
  colorAlerta: "azul",
  textoCuerpo:
    "Para mantener la seguridad de tu cuenta, por favor permite la geolocalización mientras usas la app.",
  etiquetaBoton: "Permitir",
  etiquetaBoton2: "No permitir",
  temaBoton2: "simple",
};
const configAlertaErrorGeolocation = {
  textoEncabezado: "Geolocalización",
  tipoIcono: "ubicacion",
  colorAlerta: "rojo",
  textoCuerpo: "Es necesario permitir la geolocalización para usar la app.",
  etiquetaBoton: "Aceptar",
};

export { configAlertaGeolocation, configAlertaErrorGeolocation, diccionario };
