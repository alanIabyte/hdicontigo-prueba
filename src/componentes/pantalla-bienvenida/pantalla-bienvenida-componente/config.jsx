const diccionario = {
  mensajeBienvenida: "Te damos la bienvenida a",
  mensajeDeErrorDefault: "Ocurrió un error",
  mensajeAlertaContrasenaCambiada:
    "Tu contraseña ha sido actualizada con éxito.",
  mensajeIngreso: "Ingresar",
  terminos: "Términos y condiciones",
  aviso: "Aviso de privacidad",
  etiquetaReportar: "Reportar un siniestro",
  etiquetaIngresar: "Ingresar",
  etiquetaEntrar: "Entrar",
  etiquetaUsuario: "Teléfono",
  etiquetaContrasena: "Contraseña",
  linkOlvideContrasena: "Olvidé mi contraseña",
  errorUsuario: "Teléfono debe tener 10 dígitos",
  alertaLogin: {
    titulo: "Ocurrió un problema",
    // eslint-disable-next-line max-len
    cuerpo:
      "Para seguir usando la aplicación debes iniciar sesión. También puedes reportar un siniestro sin iniciar sesión.",
    boton: "Continuar",
  },
  mensajeRegistro: "Regístrate aquí",
  mensajeUsuarioContraseniaIncorrectos: "Usuario o Contraseña incorrectos",
  mensajeHuella: "Regístra Huella",
  etiquetaEntrarHuella: "Entrar",
  etiquetaEntrarHuella2: "Usar Biometricos",
  mensajeBienvenido: "¡Bienvenido!",
};

const configAlertaGeolocation = {
  textoEncabezado: "Geolocalizacion",
  tipoIcono: "ubicacion",
  colorAlerta: "azul",
  textoCuerpo:
    "Para mantener la seguridad de tu cuenta, por favor permite la geolocalización mientras usas la app.",
  etiquetaBoton: "Permitir",
  etiquetaBoton2: "No permitir",
  temaBoton2: "simple",
};
const configAlertaErrorGeolocation = {
  textoEncabezado: "Geolocalización",
  tipoIcono: "ubicacion",
  colorAlerta: "rojo",
  textoCuerpo: "Es necesario permitir la geolocalización para usar la app.",
  etiquetaBoton: "Aceptar",
};

const configAlertaNuevosServicios = {
  textoEncabezado: "¡Bienvenido!",
  tipoIcono: "contigo",
  colorAlerta: "azul",
  textoCuerpo:
    "Te informamos que los servicios de HDI TU COMPAÑÍA estarán disponibles en HDI Contigo.",
  textoCuerpoJsx: "Navega en ella para conocer más.",
  etiquetaBoton: "Aceptar",
};

const configAlertaBometricosNoRegistrados = {
  textoEncabezado: "Biométricos aun no registrados",
  tipoIcono: "trianguloAlerta",
  colorAlerta: "rojo",
  textoCuerpo:
    "En este momento se registrarán tus datos biometricos. Para continuar con este proceso es necesario verificar tu contraseña, ¿Estás de acuerdo de continuar con este proceso?",
  etiquetaBoton: "Continuar",
  etiquetaBoton2: "En otro momento",
};

const configLocalStorage = {
  usuario: "usuario",
  contrasenia: "contrasenia",
  codigoId: "codigoId",
};

export {
  configAlertaGeolocation,
  configAlertaErrorGeolocation,
  diccionario,
  configAlertaNuevosServicios,
  configAlertaBometricosNoRegistrados,
  configLocalStorage,
};
