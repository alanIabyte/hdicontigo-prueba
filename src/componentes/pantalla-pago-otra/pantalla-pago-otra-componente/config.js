const diccionario = {
  titulo: "Otras formas de pago",
  descripcionPantalla: "Tienes más opciones para hacer tu pago",
  descargarFormato: "Descargar formato de pago",
  mensajeDeErrorDefault:
    "Tu solicitud no pudo ser procesada, por favor contacta a tu agente.",
  formas: {
    bancos: "Bancos / Cajeros automáticos",
    tiendas: "Tiendas de autoservicio",
  },
};

const configAlertaError = {
  textoEncabezado: "Ocurrió un error",
  tipoIcono: "trianguloAlerta",
  colorAlerta: "amarillo",
  textoCuerpoJsx:
    "Tu solicitud no pudo ser procesada, por favor contacta a tu agente.",
  etiquetaBoton: "Aceptar",
};

export { configAlertaError, diccionario };
