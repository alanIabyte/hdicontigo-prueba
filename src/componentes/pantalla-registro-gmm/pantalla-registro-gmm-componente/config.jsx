// Config for screen
// Por si acaso
const diccionario = {
  encabezado: "Registro",
  titulo: "Ayúdanos a registrarte correctamente",
  mensajePequeno: "Estos datos nos ayudarán a registrar tu póliza de seguro",
  etiquetaPoliza: "Número de póliza (incluyendo guiones)",
  etiquetaPolizaGMM: "Número de póliza",
  etiquetaSerie: "Últimos 4 dígitos del número de serie de tu vehículo:",
  etiquetaRFC: "RFC",
  etiquetaTelefono: "Teléfono",
  etiquetaContrasenia: "Contraseña",
  etiquetaConfirmarContrasenia: "Confirmar contraseña",
  etiquetaInciso: "Inciso",
  marcadorPoliza: "Marcador opcional",
  etiquetaBotonCompletarRegistro: "Completar registro",
  etiquetaValidar: "Verificar campos",
  linkInstruccionesPoliza: "¿Dónde encuentro el número de póliza?",
  intruccionesPoliza:
    "Localiza tu número de póliza tal como se ve en el ejemplo",
  linkInstruccionesPolizaDAN: "¿Dónde encuentro el número de póliza e inciso?",
  linkInstruccionesPolizaGMM: "¿Dónde encuentro el número de póliza e inciso?",
  linkInstruccionesSerie: "¿Dónde está mi número de serie?",
  intruccionesSerie: "Localiza tu número de serie tal como se ve en el ejemplo",
  linkInstruccionesRFC: "¿Dónde encuentro mi RFC?",
  mensajeDeErrorDefault:
    "OTu solicitud no pudo ser procesada, por favor contacta a tu agente.",
  validarCampos: "Validar campos",
  alerta: {
    titulo: "Póliza no vigente",
    cuerpo:
      // eslint-disable-next-line max-len
      "La póliza que intentas reportar presenta problemas de vigencia.<br /><br />No te preocupes, comunícate con nosotros para continuar tu atención.",
    etiquetaBoton: "Llamar a HDI *434",
  },
  errores: {
    verificarCampos:
      "Favor de verificar tus datos y que estás seleccionando el tipo de póliza correcta.",
    errorRegistro: "Ocurrio un problema al generar el usuario.",
    errorGenerico: "Hubo un error, inténtalo de nuevo más tarde",
    campoRequerido: "Campo requerido para poder continuar.",
    formatoIncorrecto: "Por favor revisa que el formato sea correcto.",
    contrasenaNoCoincide: "Las contraseñas no coinciden.",
    contrasenaNumCaracteres:
      "La contraseña debe contener al menos 8 caracteres.",
    contrasenaReglas:
      "La contraseña debe contener mayúsculas, minúsculas, números y caracteres especiales",
    rfcIncompleto: "Tu RFC debe contener al menos 12 caracteres.",
    telefonoIncompleto: "Introduce tu teléfono a 10 dígitos.",
  },
};

const configAlertaGeolocation = {
  textoEncabezado: "Geolocalizacion",
  tipoIcono: "ubicacion",
  colorAlerta: "azul",
  textoCuerpo:
    "Para mantener la seguridad de tu cuenta, por favor permite la geolocalización mientras usas la app.",
  etiquetaBoton: "Permitir",
  etiquetaBoton2: "No permitir",
  temaBoton2: "simple",
};
const configAlertaErrorGeolocation = {
  textoEncabezado: "Geolocalización",
  tipoIcono: "ubicacion",
  colorAlerta: "rojo",
  textoCuerpo: "Es necesario permitir la geolocalización para usar la app.",
  etiquetaBoton: "Aceptar",
};

export { configAlertaGeolocation, configAlertaErrorGeolocation, diccionario };
