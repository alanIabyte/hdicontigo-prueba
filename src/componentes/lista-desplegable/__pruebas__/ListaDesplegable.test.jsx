/* eslint-disable react/jsx-props-no-spreading */
import React from "react";
import renderer from "react-test-renderer";
import ListaDesplegable from "../lista-desplegable-componente/ListaDesplegable";
import config from "../__simulaciones__/ListaDesplegable.simulacion";

describe("Lista Desplegable", () => {
  it("se despliega correctamente", () => {
    const tree = renderer.create(<ListaDesplegable config={config} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
