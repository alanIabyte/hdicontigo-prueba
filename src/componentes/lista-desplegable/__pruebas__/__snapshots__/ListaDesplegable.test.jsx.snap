// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lista Desplegable se despliega correctamente 1`] = `
<div
  className="sc-kpDqfm sc-ikkxIA bTCkFB gKSbxM"
>
  <div
    className="sc-kpDqfm sc-jsJBEP bTCkFB fPA-DAX"
  >
    
  </div>
  <div
    className="sc-feUZmu enwbgs"
    onClick={[Function]}
  >
    <div
      className="sc-dAbbOL"
    >
      <div
        className="sc-fUnMCh fzoNkI"
      >
        test 1
      </div>
      <div
        className="sc-hmdomO leQpvF"
      >
        <div
          className="sc-bXCLTC eKSzWW"
        />
      </div>
    </div>
  </div>
  <div
    className="sc-hzhJZQ dpKQE"
  >
    <div
      className="sc-fHjqPf kRRpiB"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      test 1
    </div>
    <div
      className="sc-fHjqPf gFJWev"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={1}
    >
      test 2
    </div>
  </div>
</div>
`;
